### PR TITLE
Add recipe for O² Control and Configuration plugin (Control-OCCPlugin)

### PR DESCRIPTION
**Pull request by teo**
The source directory is [`occplugin`](https://github.com/AliceO2Group/Control/tree/master/occplugin) in [AliceO2Group/Control](https://github.com/AliceO2Group/Control).

### CODE REVIEW

File: control-occplugin.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# Dependencies
+module load BASE/1.0 \\
+            ${BOOST_VERSION:+boost/$BOOST_VERSION-$BOOST_REVISION} \\
+            ${GCC_TOOLCHAIN_ROOT:+GCC-Toolchain/$GCC_TOOLCHAIN_VERSION-$GCC_TOOLCHAIN_REVISION} \\
+            ${PROTOBUF_VERSION:+protobuf/$PROTOBUF_VERSION-$PROTOBUF_REVISION} \\
+            ${FAIRROOT_VERSION:+FairRoot/$FAIRROOT_VERSION-$FAIRROOT_REVISION} \\
+            ${GRPC_VERSION:+grpc/$GRPC_VERSION-$GRPC_REVISION}
+
+# Our environment
+setenv Control_OCCPlugin_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
```
Comment (ktf) [inline control-occplugin.sh]: I would use set, not setenv.

**Comment by ktf** [comment]
Can you check here:

https://github.com/alisw/alidist/pull/1124/files

and do the same? You need also to adapt the part where you use the variable.

File: protobuf.sh [hunk 2]
```diff
@@ -1,12 +1,12 @@
 package: protobuf
-version: v2.6.1
```
[the remark this answers is not in the document]
Comment (ktf) [reply protobuf.sh]: If you did not check or if you do not want to check, you should adapt it in the defaults only, I guess.

[the remark this answers is not in the document]
Comment (teo) [reply protobuf.sh]: @ktf Admittedly I don't have a deep understanding of aliBuild, but from what I can tell all the defaults that contain protobuf used to specify protobuf 3.0.2, and I've bumped them to 3.5.2. The protobuf recipe itself used to declare 2.6.1 but its check was for 3.0.0 or later. Is this a situation which allows us to get rid of the 3.5.2 override in each default?

[the remark this answers is not in the document]
Comment (ktf) [reply protobuf.sh]: The check is for < 3.0.0, i.e. "2.x".

I checked. I think the mesos framework we use for release validation can only work with protobuf 2.x at the moment. So I would not bump it in general.

The other package which right now requires it is flpproto. Do you always specify a `--defaults` for that? @Barthelemy

**Comment by ktf** [comment]
Please also add the proper entry in the CODEOWNERS file.

File: control-occplugin.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  make ${JOBS+-j $JOBS} prefix=$INSTALLROOT
+  make prefix=$INSTALLROOT install
+  mkdir -p $INSTALLROOT/etc/modulefiles && rsync -a --delete etc/modulefiles/ $INSTALLROOT/etc/modulefiles
+---
+#!/bin/bash -e
+
+LIBEXT=so
+case $ARCHITECTURE in
+    osx*)
+      [[ ! $BOOST_ROOT ]] && BOOST_ROOT=$(brew --prefix boost)
```
Comment (ktf) [inline control-occplugin.sh]: You should do the same also for protobuf and grpc since they can all come from the system.

**Comment by ktf** [comment]
Will merge when tests are ok.

**Comment by ktf** [comment]
Actually, no, since nothing uses this for the time being. Merging.

**Pull request merged**